File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" +"); // + because every type of space remove.
        
        StringBuilder sb = new StringBuilder();
        
        for(int i=words.length-1; i>=0; i--){
            sb.append(words[i]);
            sb.append(" ");
        }
        return sb.toString().trim();
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let arr = s.split(' ');
    
    let reverseStr = '';
    
    for(let i=arr.length-1; i>=0; i--){
        if(arr[i]=='') continue;
        if(reverseStr.length>0) reverseStr+=' ';
        reverseStr+=arr[i];
    }
    
    return reverseStr;
};